rbd/rbd.go

```diff
@@ -1019,6 +1019,69 @@ func (image *Image) GetParentInfo(p_pool, p_name, p_snapname []byte) error {
 	}
 }
 
+// int rbd_metadata_get(rbd_image_t image, const char *key, char *value, size_t *vallen)
+func (image *Image) GetMetadata(key string) (string, error) {
+	if err := image.validate(imageIsOpen); err != nil {
+		return "", err
+	}
+
+	c_key := C.CString(key)
+	defer C.free(unsafe.Pointer(c_key))
+
+	var c_vallen C.size_t
+	ret := C.rbd_metadata_get(image.image, c_key, nil, (*C.size_t)(&c_vallen))
+	// get size of value
+	// ret -34 because we pass nil as value pointer
+	if ret != 0 && ret != -34 {
```

Comment:
Instead of `-34`, please use `-C.ERANGE` as that matches the error from `errno.h` and is more readable.

Reply:
Thanks, for the advice